spec/jobs/application_job_spec.rb

```diff
@@ -2,34 +2,78 @@
 
 require 'rails_helper'
 
+# A very simple job class that extends ApplicationJob to allow testing of basic queue locking behavior
+class RegularParameterJob < ApplicationJob
+  def self.lock_timeout
+    1
+  end
+
+  def perform(param_a, param_b, should_raise)
+    Rails.logger.info("simulate a running job to better test queue locking: (#{param_a}, #{param_b}, #{should_raise})")
+
+    raise 'oops' if should_raise
+  end
+end
+
 describe ApplicationJob, type: :job do
+  include ActiveJob::TestHelper
+
   around do |example|
-    old_adapter = described_class.queue_adapter
-    described_class.queue_adapter = :resque
+    old_adapter = ActiveJob::Base.queue_adapter
+    ActiveJob::Base.queue_adapter = :resque
     example.run
-    described_class.queue_adapter = old_adapter
+    ActiveJob::Base.queue_adapter = old_adapter
   end
 
   it 'queues start empty' do
-    expect(Resque.info).to include(pending: 0)
+    expect(enqueued_jobs.size).to eq 0
   end
 
   context 'a subclass with message(s) queued' do
-    before do
-      allow(ZipmakerJob).to receive(:perform_later).and_call_original # undo rails_helper block
-      ZipmakerJob.perform_later('1234abc', 1)
-    end
```

Comment:
i thought this was clearer inline in the individual tests, and no need to undo zipmaker monkeypatching from rails_helper with the new test class